flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/OneSlotPerExecutionSlotAllocatorTest.java

```diff
@@ -224,6 +224,7 @@ public void testCoLocationConstraintThrowsException() {
 		final List<ExecutionVertexSchedulingRequirements> schedulingRequirements = Collections.singletonList(
 			new ExecutionVertexSchedulingRequirements.Builder()
 				.withExecutionVertexId(new ExecutionVertexID(new JobVertexID(), 0))
+				.withSlotSharingGroupId(new SlotSharingGroupId())
```

Comment:
why not to add `new SlotSharingGroupId()` as a default value of `slotSharingGroupId` in `ExecutionVertexSchedulingRequirements.Builder`?

Reply:
In production code path, a non-null `SlotSharingGroupId` should always be obtainable from the vertex.
A random `SlotSharingGroupId` is only needed in tests so I'd like to avoid to create it here. This is similar to how we deal with `ExecutionVertexID`.

Reply:
ok, a bit strange to have the builder in production code where it is used only once and the builder does not have default values for all fields.

Reply:
Agreed. But I think we will be able to drop it when we drops all other `ExecutionSlotAllocator` except for `SlotSharingExecutionSlotAllocator`.